mailbox/backup/src/main/java/org/apache/james/mailbox/backup/DefaultMailboxBackup.java

```diff
@@ -0,0 +1,167 @@
+            return messages;
+        }
+    }
+
+    public DefaultMailboxBackup(MailboxManager mailboxManager, ArchiveService archiveService, MailArchivesLoader archiveLoader, BlobStore store) {
+        this.mailboxManager = mailboxManager;
+        this.archiveService = archiveService;
+        this.archiveLoader = archiveLoader;
+        this.store = store;
+    }
+
+    @Override
+    public Publisher<BlobId> backupAccount(User user) throws IOException, MailboxException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        List<MailAccountContent> accountContents = getAccountContentForUser(session);
+        List<MailboxWithAnnotations> mailboxes = accountContents.stream()
+            .map(MailAccountContent::getMailboxWithAnnotations)
+            .collect(Guavate.toImmutableList());
+
+        Stream<MessageResult> messages = allMessagesForUser(accountContents);
+        return saveToStore(mailboxes, messages);
+    }
+
+    @Override
+    public Publisher<Void> restore(User user, BlobId blobId) throws IOException, MailboxException {
+        MailboxSession session = mailboxManager.createSystemSession(user.asString());
+        org.apache.james.blob.api.BlobId.Factory blobIdFactory = new HashBlobId.Factory();
+        try (InputStream in = store.read(blobIdFactory.from(blobId.asString()));
+             MailArchiveIterator archiveIterator = archiveLoader.load(in)) {
+            throw new NotImplementedException("TODO");
+        }
+    }
+
+    private final MailboxManager mailboxManager;
+    private final ArchiveService archiveService;
+    private final MailArchivesLoader archiveLoader;
+    private final BlobStore store;
+
+    private Function<MailboxPath, Optional<MailAccountContent>> getMailboxWithAnnotationsFromPath(MailboxSession session) {
+        return path -> {
+            try {
+                StoreMessageManager messageManager = (StoreMessageManager) mailboxManager.getMailbox(path, session);
+                Mailbox mailbox = messageManager.getMailboxEntity();
+                List<MailboxAnnotation> annotations = mailboxManager.getAllAnnotations(path, session);
+                MailboxWithAnnotations mailboxWithAnnotations = new MailboxWithAnnotations(mailbox, annotations);
+                Stream<MessageResult> messages = Iterators.toStream(messageManager.getMessages(MessageRange.all(), FetchGroupImpl.FULL_CONTENT, session));
+                return Optional.of(new MailAccountContent(mailboxWithAnnotations, messages));
+            } catch (MailboxException e) {
+                LOGGER.error("Error while fetching Mailbox during backup", e);
+                return Optional.empty();
+            }
+        };
+    }
+
+    @VisibleForTesting
+    List<MailAccountContent> getAccountContentForUser(MailboxSession session) throws MailboxException {
+        MailboxQuery queryUser = MailboxQuery.builder().username(session.getUser().asString()).build();
```

Comment:
This `.username(session.getUser().asString())` is dirty. Please add a `.user(session.getUser())`...